tcslackbuildnotifier-core/src/main/java/slacknotifications/SlackNotificationImpl.java

```diff
@@ -368,62 +369,43 @@ private void sendToChannel(String channel, WebHookPayload requestBody, String ur
     //Get commits from the user being sent to
     //Divided into own section away from rest of commits
     StringBuilder sbUserCommits = new StringBuilder();
-    int numUserCommits = 0;
-    int numUserCommitsLeft = 0;
-    int numCommitsToAdd = maxCommitsToDisplay;
 
     List<Commit> commits = this.payload.getCommits();
 
-    List<Commit> commitsToDisplay = new ArrayList<Commit>(commits);
-
-    if (showCommits) {
-      boolean truncatedOther = false;
-      boolean truncatedUser = false;
-      int totalCommits = commitsToDisplay.size();
-
-      //determine how many commits were from the user
-      //used to determine how many in "Your commits" and how many in "Commits"
-      for (Commit commit : commitsToDisplay) {
-        if (commit.getSlackUserId().equals(slackUser.substring(1))) {
-          numUserCommits++;
-        }
-      }
-      numUserCommitsLeft = numUserCommits;
+    List<Commit> userCommits = new LinkedList<Commit>();
+    List<Commit> otherCommits = new LinkedList<Commit>();
 
-      if (numUserCommits > maxCommitsToDisplay) {
-        truncatedUser = true;
-      }
-
-      if (totalCommits > maxCommitsToDisplay && totalCommits > numUserCommits) {
-        //commitsToDisplay = commitsToDisplay.subList(0, maxCommitsToDisplay > commitsToDisplay.size() ? commitsToDisplay.size() : 5);
-        truncatedOther = true;
+    // Separate user commits from other commits
+    for(Commit commit : commits){
+      if (commit.getSlackUserId().equals(slackUser.substring(1))) {
```

Comment:
Could `commit.getSlackUserId()` not return a name that starts with `@`? Plus it is not clear from this line why one needs to use `substring(1)`. It would be good to standardize the slack user id format. Consider putting logic into the `Commit` class such that it returns a consistent slack user id format.

Reply:
Does `commit.getSlackUserId` necessarily return a String, or could it return null? If so, flip the equals logic to avoid NPE:
`slackUser.substring(1).equals(commit.getSlackUserId())`